Tchap/Modules/Application/AppCoordinator.swift

```diff
@@ -54,6 +58,79 @@ final class AppCoordinator: AppCoordinatorType {
         }
     }
     
+    func handleUserActivity(_ userActivity: NSUserActivity, application: UIApplication) -> Bool {
+        if userActivity.activityType == NSUserActivityTypeBrowsingWeb {
+            return self.universalLinkService.handleUserActivity(userActivity, completion: { (response) in
+                switch response {
+                case .success(let parsingResult):
+                    switch parsingResult {
+                    case .registrationLink(let registerParams):
+                        self.handleRegisterAfterEmailValidation(registerParams)
+                    case .roomLink(let roomIdOrAlias, let eventID):
+                        _ = self.showRoom(with: roomIdOrAlias, onEventID: eventID)
+                    }
+                case .failure(let error):
+                    // FIXME: Present an error on coordinator.toPresentable()
+                    AppDelegate.theDelegate().showError(asAlert: error)
+                }
+            })
+        } else if userActivity.activityType == INStartAudioCallIntentIdentifier ||
+        userActivity.activityType == INStartVideoCallIntentIdentifier {
+            // Check whether a session is available (Ignore multi-accounts FTM)
+            guard let account = MXKAccountManager.shared()?.activeAccounts.first else {
+                return false
+            }
+            guard let session = account.mxSession else {
+                return false
+            }
+            let interaction = userActivity.interaction
+            
+            // Check roomID provided by Siri intent
+            var roomID = userActivity.userInfo?["roomID"] as? String
```

Comment:
Prefer `if let roomID = userActivity.userInfo?["roomID"] as? String {` and show an error if roomID is mandatory below

Reply:
I don't understand your remark, did you notice that we check `roomID == nil` and not != nil in the following block?
But I understood I should rewrite this code to use `let roomID`instead of `var roomID`

Reply:
Yes I would say write something like:
```
let finalRoomID: String?

if let roomID = userActivity.userInfo?["roomID"] as? String {
    finalRoomID = roomID
} else {
    let person: INPerson?
    
    if let audioCallIntent = interaction?.intent as? INStartAudioCallIntent {
        person = audioCallIntent.contacts?.first
    } else if let videoCallIntent = interaction?.intent as? INStartVideoCallIntent {
        person = videoCallIntent.contacts?.first
    } else {
        person = nil
    }
    
    finalRoomID = person?.personHandle?.value
}
```

Reply:
Below there is a call made on a possible empty room id:
`session.callManager.placeCall(inRoom: roomID ?? "", withVideo: isVideoCall, success: { (call) in`
It would be preferable to display an error before making the call in this case.

Reply:
done